setuptools/msvc.py

```diff
@@ -525,9 +528,24 @@ def VCInstallDir(self):
         """
         Microsoft Visual C++ directory.
         """
-        # Default path
-        default = r'Microsoft Visual Studio %0.1f\VC' % self.vc_ver
-        guess_vc = os.path.join(self.ProgramFilesx86, default)
+        self.VSInstallDir
+
+        # Default path starting VS2017
+        guess_vc = ''
+        if self.vc_ver > 14.0:
+            default = r'VC\Tools\MSVC'
+            guess_vc = os.path.join(self.VSInstallDir, default)
+            # Subdir with VC exact version as name
+            try:
+                vc_exact_ver = os.listdir(guess_vc)[-1]
```

Comment:
This literal `-1` makes me a little uneasy. What's happening here? Is it assuming that listdir is returning the directories sorted alphanumerically and that the latest version will also be the largest alphanumeric? What sorts of values does one expect to find in os.listdir(guess_vc)?